src/ucp/rndv/rndv_ppln.c

```diff
@@ -204,14 +205,11 @@ static ucs_status_t ucp_proto_rndv_ppln_progress(uct_pending_req_t *uct_req)
         ucp_trace_req(req, "send fragment request %p", freq);
         ucp_request_send(freq);
 
-        if (ucp_datatype_iter_is_end_position(&req->send.state.dt_iter,
-                                              &next_iter)) {
-            return UCS_OK;
-        }
-
         ucp_datatype_iter_copy_position(&req->send.state.dt_iter, &next_iter,
                                         UCS_BIT(UCP_DATATYPE_CONTIG));
     }
+
+    return UCS_OK;
```

Comment:
btw, is there a case when need to return not ok? Maybe make it void?

Reply:
it's a UCT progress function, so it must return status